start-site/src/test/java/io/spring/start/site/extension/SpringCloudContractProjectContributorTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.start.site.extension;
+
+import io.spring.initializr.generator.test.project.ProjectStructure;
+import io.spring.initializr.web.project.ProjectRequest;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link SpringCloudContractProjectContributor}.
+ *
+ * @author Eddú Meléndez
+ */
+class SpringCloudContractProjectContributorTests extends AbstractExtensionTests {
+
+	@Test
+	void contractsDirectoryIsCreatedWithSpringCloudContractVerifier() {
```

Comment:
I don't know what this `web` component means, but you can have contracts for a messaging application that doesn't have a web interface.

Reply:
this test makes sure that when `cloud-contract-verifier` dependency is added `contracts` folder is created. it doesn't matter if it is a web application or not.

Reply:
To clarify things, `"web"` could be removed and the test will still be valid.